### PR TITLE
Replace strtok in systemd-sonic-generator

### PR DESCRIPTION
**Pull request by maipbui**
Signed-off-by: maipbui <maibui@microsoft.com>

#### Why I did it
Replace unsafe functions to safe functions
#### How I did it
Replace `strtok()` by `strtok_r()`
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

**Review by qiluo-msft — APPROVED** [review]

**Comment by qiluo-msft** [comment]
@SuvarnaMeenakshi @theasianpianist Could you help review?

File: src/systemd-sonic-generator/systemd-sonic-generator.c [hunk 1]
```diff
@@ -547,8 +550,8 @@ int get_num_of_asic() {
         if (fp != NULL) {
             while ((nread = getline(&line, &len, fp)) != -1) {
                 if (strstr(line, "NUM_ASIC") != NULL) {
-                    token = strtok(line, "=");
-                    str_num_asic = strtok(NULL, "=");
+                    token = strtok_r(line, "=", &saveptr2);
```
Comment (SuvarnaMeenakshi) [inline src/systemd-sonic-generator/systemd-sonic-generator.c:553]: can we use the same saveptr? instead of two separate pointers?

Comment (maipbui) [reply src/systemd-sonic-generator/systemd-sonic-generator.c]: I think we can. Fixed to one pointer.

**Review by theasianpianist — APPROVED** [review]

**Review by SuvarnaMeenakshi — APPROVED** [review]
lgtm

**Pull request merged**